SimPPS/RPDigiProducer/plugins/RPDisplacementGenerator.cc

```diff
@@ -16,34 +12,37 @@
 using namespace std;
 using namespace edm;
 
-RPDisplacementGenerator::RPDisplacementGenerator(const edm::ParameterSet &ps,
+RPDisplacementGenerator::RPDisplacementGenerator(bool iIsOn,
                                                  RPDetId _detId,
-                                                 const edm::EventSetup &iSetup)
+                                                 const CTPPSRPAlignmentCorrectionsData *alignments,
+                                                 const CTPPSGeometry &geom)
     : detId_(_detId) {
-  isOn_ = ps.getParameter<bool>("RPDisplacementOn");
+  isOn_ = iIsOn;  //ps.getParameter<bool>("RPDisplacementOn");
 
   // read the alignment correction
+  /*
   ESHandle<CTPPSRPAlignmentCorrectionsData> alignments;
   if (auto rec = iSetup.tryToGet<VeryForwardMisalignedGeometryRecord>()) {
     iSetup.get<VeryForwardMisalignedGeometryRecord>().get(alignments);
   }
+  */
 
   unsigned int decId = rawToDecId(detId_);
 
   math::XYZVectorD S_m;
   RotationMatrix R_m;
 
-  if (alignments.isValid()) {
+  if (alignments) {
     const CTPPSRPAlignmentCorrectionData &ac = alignments->getFullSensorCorrection(decId);
     S_m = ac.getTranslation();
     R_m = ac.getRotationMatrix();
   } else
     isOn_ = false;
 
   // transform shift and rotation to the local coordinate frame
-  ESHandle<CTPPSGeometry> geom;
-  iSetup.get<VeryForwardRealGeometryRecord>().get(geom);
-  const DetGeomDesc *g = geom->sensor(detId_);
+  //ESHandle<CTPPSGeometry> geom;
```

Comment:
Is this supposed to remain commented out? Or can we profit and remove these lines now?

Reply:
done